JSDemos/Demos/TreeList/RowEditing/description.md

```diff
@@ -1 +1,5 @@
-This example demonstrates editing in _&laquo;row&raquo;_ mode. Each row is&nbsp;supplied with an _&laquo;Edit&raquo;_ link, a&nbsp;click on&nbsp;which switches the row into the editing state. Editors vary from one cell to&nbsp;another: they may be&nbsp;calendars, select boxes, simple text boxes, etc. Note that you can also add or&nbsp;delete a&nbsp;row using the _&laquo;Add a&nbsp;row&raquo;_ button or&nbsp;the _&laquo;Delete&raquo;_ link, respectively.
\ No newline at end of file
+To enable the row editing mode, set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"row"*.
+
+All edit operations are deactivated in the default configuration. To activate them, set the [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties of the editing object to **true**. This will create the edit column with buttons for each allowed operation on each row. In this demo, we declared a column of type *"buttons"* to add the edit and delete buttons only; the universal button to add a new row is situated in the upper right corner of the demo container. See the [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column) article for details.
+
+Each edit operation raises events. In this demo, we use the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler to configure the creation of a new row: the Head of a new employee is prepopulated with the company's CEO. We also use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize the edit form: if a user edits the row with the CEO, the TreeList will disable the Head column editor.
```

Comment:
1. >We also use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize the edit form

It seems to me that we should list the existing events and their purpose and only then show that these events are used for. Moreover, there's no form in this editing mode.

2. We should add a link to the documentation section that lists all events raised by editing.

```suggestion
Each edit operation raises events. The [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates an added row with data. In this demo, this handler assigns the CEO's `ID` to the `Head` field of a newly created row. Use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize a cell's editor. In this demo, we use this handler to disable an editor for the `Head` field when users edit the row with the CEO.

Refer to the following help topic to see the full list of the events raised by editing: [Editing Events](/Documentation/Guide/UI_Components/TreeList/Editing/#Events).
```